internal/vpn/net.go

```diff
@@ -0,0 +1,52 @@
+package vpn
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+)
+
+// WriteJSON marshals `data` and sends it over the `conn`.
+func WriteJSON(conn net.Conn, data interface{}) error {
+	dataBytes, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("error marshaling data: %w", err)
+	}
+
+	for n, totalSent := 0, 0; totalSent < len(dataBytes); totalSent += n {
+		n, err = conn.Write(dataBytes[totalSent:])
+		if err != nil {
+			return fmt.Errorf("error sending data: %w", err)
+		}
+
+		totalSent += n
+	}
+
+	return nil
+}
+
+// ReadJSON reads portion of data from the `conn` and unmarshals it into `data`.
+func ReadJSON(conn net.Conn, data interface{}) error {
+	const bufSize = 1024
+
+	var dataBytes []byte
+	buf := make([]byte, bufSize)
+	for {
+		n, err := conn.Read(buf)
```

Comment:
Could `ioutil.ReadAll` be used here instead?

Reply:
@nkryuchkov unfortunately it won't work. `ReadAll` keeps reading until `EOF` or some other error occurs. but in our case `EOF` won't happen, since it rises only if remote got closed or empty data received from it. which never happens because of how route group is implemented

Reply:
@Darkren I see. Thank you for the explanation.